.github/workflows/build_and_test.yml

```diff
@@ -96,97 +108,148 @@ jobs:
-      if: matrix.os == 'ubuntu-22.04' && (needs.check_changes.outputs.tket_changed == 'true' || needs.check_changes.outputs.tket_tests_changed == 'true') && github.event_name == 'schedule'
+      if: matrix.os == 'ubuntu-22.04' && github.event_name == 'schedule'
       run: |
         sudo apt install texlive texlive-latex-extra latexmk
         mkdir -p ~/texmf/tex/latex
         wget http://mirrors.ctan.org/graphics/pgf/contrib/quantikz/tikzlibraryquantikz.code.tex -P ~/texmf/tex/latex
-    - name: Build and run tket tests
-      if: needs.check_changes.outputs.tket_changed == 'true' || needs.check_changes.outputs.tket_tests_changed == 'true'
-      shell: bash
-      run: |
-        conan create tket/test --build=missing -o boost/*:header_only=True --format json > test-tket.json
-        PKGPATH=`./rootpath test-tket.json test-tket`
-        cd ${PKGPATH}/bin && ./test-tket `./github/workflows/test-args ${{ matrix.os }} ${{ github.event_name }}`
-    - name: Build and run tket proptests
-      if: needs.check_changes.outputs.tket_changed == 'true' || needs.check_changes.outputs.tket_proptests_changed == 'true'
-      shell: bash
-      run: |
-        conan create tket/proptest --build=missing  -o boost/*:header_only=True --format json > proptest-tket.json
-        PKGPATH=`./rootpath proptest-tket.json proptest-tket`
-        cd ${PKGPATH}/bin && ./proptest-tket
+    - name: ccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        key: tket-static-${{ matrix.os }}
+    - name: further ccache config
+      run: |
+        ccache --set-config base_dir=${HOME}
+        ccache --set-config compiler_check=content
+        ccache --set-config namespace=WITH_TESTS
+        ccache -p
+    - name: Build tket
+      run: conan create tket -s build_type=Release --user=tket --channel=stable -o boost/*:header_only=True -o with_all_tests=True
     - name: Upload package
-      if: github.event_name == 'push' && github.ref == 'refs/heads/develop' && needs.check_changes.outputs.tket_changed == 'true'
+      if: github.event_name == 'push' && github.ref == 'refs/heads/develop'
       run: |
+        ccache --set-config namespace=WITHOUT_TESTS
+        conan create tket --user=tket --channel=stable -o boost/*:header_only=True
         conan remote login -p ${{ secrets.JFROG_ARTIFACTORY_TOKEN_2 }} tket-libs ${{ secrets.JFROG_ARTIFACTORY_USER_2 }}
         conan upload tket/${{ needs.check_changes.outputs.tket_ver }}@tket/stable -r=tket-libs
 
+  build_test_tket_windows:
+    name: Build and test (tket)
+    needs: check_changes
+    if: needs.check_changes.outputs.tket_changed == 'true'
+    strategy:
+      matrix:
+        os: ['windows-2022']
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Select Python 3.10
+        # otherwise turtlebrowser/get-conan@v1.2 fails on macos-12
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```

Comment:
This probably isn't necessary on Windows. (Not even sure it's still an issue on MacOS.)